jobs/director/templates/nginx_ctl

```diff
@@ -26,8 +30,22 @@ case $1 in
 
   stop)
     PID=$(head -1 $PIDFILE)
-    kill $PID
-    while [ -e /proc/$PID ]; do sleep 0.1; done
+    if [ ! -z $PID ] && pid_exists $PID; then
+      kill $PID
+    fi
+    TRIES=0
+    while [ -e /proc/$PID ]
+    do
+      TRIES=$(( $TRIES + 1 ))
+      if [ $TRIES -gt 100 ]; then
+        CHILD_PIDS=$(ps --forest -o pid=  -g $(ps -o sid= $PID) | sed 's/ //g' | grep -v "^${PID}$")
```

Comment:
The `--forest` isn't needed here, and this command relies on nginx always creating its children in such a way that they have the same session id as the master process. 

Since the stemcells install the [`psmisc`](http://psmisc.sourceforge.net/) package, we have access to the pstree utility. we could instead use something like `pstree -p 679 | grep -E "\([0-9]+\)" --only-matching | tr --delete "()"`, which would ensure that we cover both the nested children case and the differing sid/gid case.

Reply:
@cppforlife @dpb587 wdyt